src/Symfony/Component/HttpKernel/EventListener/LocaleListener.php

```diff
@@ -65,6 +67,11 @@ private function setLocale(Request $request)
         if ($locale = $request->attributes->get('_locale')) {
             $request->setLocale($locale);
         }
+
+        if (null === $locale && null !== $this->availableLocales &&
+            $preferredLanguage = $request->getPreferredLanguage($this->availableLocales)) {
+            $request->setLocale($preferredLanguage);
```

Comment:
can't this lead to HTTP cache poisoning?
a `Vary` header could be required.

Reply:
it could, could it?
I'll add a vary.

Reply:
please note that:
Varnish does support the vary header.
Cloudflare cache does **NOT** respect the vary header.

https://support.cloudflare.com/hc/en-us/articles/115003206852-Understanding-Origin-Cache-Control
https://varnish-cache.org/docs/6.4/users-guide/increasing-your-hitrate.html#http-vary